lib/server_sent_event.ex

```diff
@@ -121,6 +134,52 @@ defmodule ServerSentEvent do
   defp single_line?(text) do
     length(String.split(text, @new_line, parts: 2)) == 1
   end
+  @doc """
+  Parse all events from text stream.
+
+  ## Examples
+  *In these examples this module has been aliased to `SSE`*.
+
+      iex> SSE.parse_all("data: First message\\n\\ndata: Second\\ndata: message\\n\\nrest")
+      {:ok,
+        {
+          [
+            %SSE{lines: ["First message"]},
+            %SSE{lines: ["Second", "message"]}
+          ],
+          "rest"
+        }
+      }
+
+      iex> SSE.parse_all("data: This is the first message\\n\\n")
+      {:ok, {[%SSE{lines: ["This is the first message"]}], ""}}
+
+      iex> SSE.parse_all("data: This is the first message\\n\\nrest")
+      {:ok, {[%SSE{lines: ["This is the first message"]}], "rest"}}
+
+      iex> SSE.parse_all("data: This message is not complete")
+      {:ok, {[], "data: This message is not complete"}}
+
+      iex> SSE.parse_all("This line is invalid\\nit doesn't contain a colon\\n")
+      {:error, {:malformed_line, "This line is invalid"}}
```

Comment:
Can we have an example for "data: This is the first message\\n\\nThis line is invalid\\n" 
and  "data: This is the first message\\n\\nThis line is yet to terminate"